### ISSUE TITLE
react-scripts build is failing with TS1005 for balanced-pool.d.ts type

### ISSUE THREAD

**Issue by nk-reichenbach-fall** [nodejs/undici#3441]
## Bug Description

`react-scripts build` is failing with TS1005 for balanced-pool.d.ts type

```
/node_modules/undici-types/balanced-pool.d.ts
TypeScript error in /node_modules/undici-types/balanced-pool.d.ts(22,12) : 
';' expected.  TS1005
     20 | 
     21 |   // Override dispatcher APIs.
  >  22 |   override connect(
        |            ^
     23 |     options: BalancedPoolConnectOptions
     24 |   ): Promise<Dispatcher.ConnectData>;
     25 |   override connect(
```

## Reproducible By

undici types is being installed as a dep of @types/node:

"node_modules/@types/node": {
      "version": "22.1.0",
      "resolved": "https://registry.npmjs.org/@types/node/-/node-22.1.0.tgz",
      "dev": true,
      "dependencies": {
        "undici-types": "~6.13.0"
      }
    }

## Expected Behavior

Build should pass

## Environment

node - v20.15.1,
react-scripts: "3.4.1",

**Comment by metcoder95** [comment]
Can this be reproduced in isolation without `react-scripts`?

**Comment by LucDepauw** [comment]
In our build we have a similar problem, library is also injected by @type/nodes, it's an Angular 12 app that was build a few weeks ago fine, but since friday we have this error showing up: (And no react-scripts  in the repo)

**Error:**

  ```
[INFO] ✖ Compiling TypeScript sources through NGC

  [INFO] ERROR: [96mnode_modules/undici-types/balanced-pool.d.ts[0m:[93m22[0m:[93m12[0m - 
[91merror[0m[90m TS1005: [0m';' expected.
  
[INFO] 

  [INFO] [7m22[0m   override connect(

  [INFO] [7m  [0m [91m           ~~~~~~~[0m

  [INFO] [96mnode_modules/undici-types/balanced-pool.d.ts[0m:[93m25[0m:[93m12[0m - [91merror[0m[90m TS1005: [0m';' expected.
  [INFO] 
```

**Environment:**
<node.version>14.17.6</node.version>
<npm.version>6.14.15</npm.version>

package.json

  ```
{
    "_from": "undici-types@~6.13.0",
    "_id": "undici-types@6.13.0",
    "_inBundle": false,
    "_integrity": "sha512-xtFJHudx8S2DSoujjMd1WeWvn7KKWFRESZTMeL1RptAYERu29D6jphMjjY+vn96jvN3kVPDNxU/E13VTaXj6jg==",
    "_location": "/undici-types",
    "_phantomChildren": {},
    "_requested": {
      "type": "range",
      "registry": true,
      "raw": "undici-types@~6.13.0",
      "name": "undici-types",
      "escapedName": "undici-types",
      "rawSpec": "~6.13.0",
      "saveSpec": null,
      "fetchSpec": "~6.13.0"
    },
    "_requiredBy": [
      "/@types/node"
    ],
```

**Comment by mcollina** [comment]
Thanks for reporting!

Can you provide steps to reproduce? We often need a [reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), e.g. some code that allows someone else to recreate your problem by just copying and pasting it. If it involves more than a couple of different file, create a new repository on GitHub and add a link to that.

**Comment by LucDepauw** [comment]
For my particular case I could solved the issue by upgrading from typescript 4.2.4 to 4.3.5, error on the undici-types gone. Did not found root cause.

**Comment by Uzlopak** [comment]
Closing due to inactivity.
